File: Python/1045 - Tipos de Triângulos.py
A,B,C = map(float,input().split(" "))

if(B>A):
    aux = B
    B = A
    A = aux

if(C>A):
    aux = C
    C = A
    A = aux

if(A>=(B+C)):
  print("NAO FORMA TRIANGULO")

elif((A**2)==(B**2)+(C**2)):
  print("TRIANGULO RETANGULO")

elif((A**2)>(B**2)+(C**2)):
  print("TRIANGULO OBTUSANGULO")

elif((A**2)<(B**2)+(C**2)):
  print("TRIANGULO ACUTANGULO")

if(A == B == C):
  print("TRIANGULO EQUILATERO")

elif((A == B) or (A == C) or (B == C)):
  print("TRIANGULO ISOSCELES")

File: Clojure/1045 - Tipos de Triângulos.clj
(let [[A B C] (map #(Double/parseDouble %) (clojure.string/split (read-line) #"\s+"))
      [A B C] (let [[x y z] [A B C]]
                (let [[A1 B1] (if (> y x) [y x] [x y])
                      [A2 B2 C2] (if (> z A1) [z B1 A1] [A1 B1 z])]
                  [A2 B2 C2]))]
  (cond
    (>= A (+ B C)) (println "NAO FORMA TRIANGULO")
    (= (* A A) (+ (* B B) (* C C))) (println "TRIANGULO RETANGULO")
    (> (* A A) (+ (* B B) (* C C))) (println "TRIANGULO OBTUSANGULO")
    (< (* A A) (+ (* B B) (* C C))) (println "TRIANGULO ACUTANGULO"))
  
  (cond
    (and (= A B) (= B C)) (println "TRIANGULO EQUILATERO")
    (or (= A B) (= A C) (= B C)) (println "TRIANGULO ISOSCELES")))
